Multisig/Data/Services/Utils/GSError.swift

```diff
@@ -0,0 +1,343 @@
+            return AppError(description: description,
+                            reason: error.reason,
+                            howToFix: error.howToFix,
+                            domain: error.domain,
+                            code: error.code,
+                            loggable: error.loggable)
+        } else if let error = error as? LocalizedError {
+            return UnknownAppError(description: description,
+                                   reason: error.failureReason ?? error.localizedDescription,
+                                   howToFix: error.recoverySuggestion ?? "")
+        } else {
+            let error = error as NSError
+            return UnknownAppError(
+                description: description,
+                reason: error.localizedFailureReason ?? error.localizedDescription,
+                howToFix: error.localizedRecoverySuggestion ?? "")
+        }
+    }
+
+    static func detailedError(from error: Error) -> Error {
+        guard let nsError = error as NSError? else { return error }
+
+        switch URLError.Code(rawValue: nsError.code) {
+        case .notConnectedToInternet:
+            return NoInternet()
+        case .secureConnectionFailed:
+            return SecureConnectionFailed()
+        case .timedOut:
+            return TimeOut()
+        case .cannotFindHost:
+            return UnknownHost()
+        default:
+            return error
+        }
+    }
+
+    static func detailedError(from httpResponse: HTTPURLResponse, data: Data?) -> Error {
+        switch httpResponse.statusCode {
+        case 200...299:
+            preconditionFailure("Not an error, please check the calling code")
+        case 404:
+            return EntityNotFound()
+        case 422:
+            return unprocessableEntity(data: data)
+        case 300...599:
+            return ServerSideError(code: httpResponse.statusCode)
+        default:
+            let error = UnknownNetworkError(code: httpResponse.statusCode)
+            LogService.shared.error("Unknown error with status code: \(httpResponse.statusCode)", error: error)
+            return UnknownNetworkError(code: httpResponse.statusCode)
+        }
+    }
+
+    private static func unprocessableEntity(data: Data?) -> Error {
+        guard let data = data else {
+            LogService.shared.error("Missing data in unprocessableEntity error", error: unexpectedError)
+            return unexpectedError
+        }
+
+        do {
+            let error = try JSONDecoder().decode(BackendError.self, from: data)
+            switch error.code {
+            case 1:
+                return UnprocessableEntity(reason: "Address format is not valid.", code: 42201)
+            case 50:
+                return UnprocessableEntity(reason: "Safe info is not found.", code: 42250)
+            default:
+                LogService.shared.error("Unrecognised error with code: \(error.code); message: \(error.message)",
```

Comment:
by the way, for all other cases the error code should have the format "422\(error.code)"

Reply:
anyway, I think it's fine